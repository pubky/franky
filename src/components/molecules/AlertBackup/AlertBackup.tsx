import * as Atoms from '@/atoms';
import * as Libs from '@/libs';
import * as Molecules from '@/molecules';

export const AlertBackup = () => {
  return (
    <Atoms.Container className="px-6 py-3 bg-brand rounded-lg flex flex-row items-center gap-3">
      <Atoms.Container className="flex flex-row flex-1 gap-3 items-center">
        <Libs.TriangleAlert className="h-4 w-4 font-bold text-primary-foreground" />
        <Atoms.Typography size="sm" className="font-bold text-primary-foreground">
          Back up now to avoid losing your account!
        </Atoms.Typography>
      </Atoms.Container>
      <Molecules.DialogBackup />
      <Molecules.DialogConfirmBackup />
    </Atoms.Container>
  );
};
