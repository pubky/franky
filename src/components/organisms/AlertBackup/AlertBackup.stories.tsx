import type { Meta, StoryObj } from '@storybook/nextjs-vite';
import { AlertBackup } from './AlertBackup';

const meta = {
  title: 'Molecules/AlertBackup',
  component: AlertBackup,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    onDismiss: { action: 'dismissed' },
  },
} satisfies Meta<typeof AlertBackup>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};

export const WithDismissHandler: Story = {
  args: {
    onDismiss: () => console.log('Dismissed'),
  },
};
