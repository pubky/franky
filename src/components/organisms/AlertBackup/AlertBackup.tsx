'use client';

import { useState, useEffect } from 'react';
import * as Atoms from '@/atoms';
import * as Libs from '@/libs';
import * as Core from '@/core';
import * as Organisms from '@/organisms';

/**
 * AlertBackup
 *
 * Self-contained component that manages its own visibility state.
 * Shows an alert when user has a secret key that needs to be backed up.
 * No props needed - manages its own state internally.
 */
export const AlertBackup = () => {
  const { secretKey } = Core.useOnboardingStore();
  const [showAlert, setShowAlert] = useState(false);

  useEffect(() => {
    if (secretKey) {
      setShowAlert(true);
    }
  }, [secretKey]);

  const handleDismiss = () => {
    setShowAlert(false);
  };

  // Don't show alert if there's no secretKey or it was dismissed
  if (!secretKey || !showAlert) {
    return null;
  }

  return (
    <Atoms.Container className="px-6 py-3 bg-brand rounded-lg flex-row items-center gap-3">
      <Atoms.Container className="flex-row flex-1 gap-3 items-center">
        <Libs.TriangleAlert className="h-4 w-4 font-bold text-primary-foreground" />
        <Atoms.Typography size="sm" className="font-bold text-primary-foreground whitespace-nowrap">
          Back up now<span className="hidden md:inline"> to avoid losing your account!</span>
        </Atoms.Typography>
      </Atoms.Container>
      <Organisms.DialogBackup />
      <Organisms.DialogConfirmBackup onConfirm={handleDismiss} />
    </Atoms.Container>
  );
};
