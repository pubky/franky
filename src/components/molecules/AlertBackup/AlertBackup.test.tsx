import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import { AlertBackup } from './AlertBackup';

// Mock atoms
vi.mock('@/atoms', () => ({
  Container: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <div data-testid="container" className={className}>
      {children}
    </div>
  ),
  Typography: ({ children, size, className }: { children: React.ReactNode; size?: string; className?: string }) => (
    <p data-testid="typography" data-size={size} className={className}>
      {children}
    </p>
  ),
}));

// Mock libs
vi.mock('@/libs', () => ({
  TriangleAlert: ({ className }: { className?: string }) => (
    <div data-testid="triangle-alert" className={className}>
      TriangleAlert
    </div>
  ),
}));

// Mock molecules
vi.mock('@/molecules', () => ({
  DialogBackup: () => <div data-testid="dialog-backup">DialogBackup</div>,
  DialogConfirmBackup: () => <div data-testid="dialog-confirm-backup">DialogConfirmBackup</div>,
}));

describe('AlertBackup', () => {
  it('renders with correct structure', () => {
    render(<AlertBackup />);

    const mainContainer = screen.getAllByTestId('container')[0];
    expect(mainContainer).toHaveClass(
      'px-6',
      'py-3',
      'bg-brand',
      'rounded-lg',
      'flex',
      'flex-row',
      'items-center',
      'gap-3',
    );
  });

  it('renders warning icon and message', () => {
    render(<AlertBackup />);

    expect(screen.getByTestId('triangle-alert')).toBeInTheDocument();
    expect(screen.getByText('Back up now to avoid losing your account!')).toBeInTheDocument();
  });

  it('renders dialog components', () => {
    render(<AlertBackup />);

    expect(screen.getByTestId('dialog-backup')).toBeInTheDocument();
    expect(screen.getByTestId('dialog-confirm-backup')).toBeInTheDocument();
  });

  it('applies correct styling to warning message', () => {
    render(<AlertBackup />);

    const warningText = screen.getByText('Back up now to avoid losing your account!');
    expect(warningText).toHaveClass('font-bold', 'text-primary-foreground');
  });

  it('applies correct styling to warning icon', () => {
    render(<AlertBackup />);

    const warningIcon = screen.getByTestId('triangle-alert');
    expect(warningIcon).toHaveClass('h-4', 'w-4', 'font-bold', 'text-primary-foreground');
  });

  it('renders all required elements', () => {
    render(<AlertBackup />);

    expect(screen.getByTestId('triangle-alert')).toBeInTheDocument();
    expect(screen.getByText('Back up now to avoid losing your account!')).toBeInTheDocument();
    expect(screen.getByTestId('dialog-backup')).toBeInTheDocument();
    expect(screen.getByTestId('dialog-confirm-backup')).toBeInTheDocument();
  });
});
