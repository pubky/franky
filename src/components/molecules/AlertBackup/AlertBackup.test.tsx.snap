// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AlertBackup > matches snapshot for default AlertBackup 1`] = `
<div
  class="px-6 py-3 bg-brand rounded-lg flex-row items-center gap-3"
  data-testid="container"
>
  <div
    class="flex-row flex-1 gap-3 items-center"
    data-testid="container"
  >
    <div
      class="h-4 w-4 font-bold text-primary-foreground"
      data-testid="triangle-alert"
    >
      TriangleAlert
    </div>
    <p
      class="font-bold text-primary-foreground whitespace-nowrap"
      data-size="sm"
      data-testid="typography"
    >
      Back up now
      <span
        class="hidden md:inline"
      >
        to avoid losing your account!
      </span>
    </p>
  </div>
  <div
    data-testid="dialog-backup"
  >
    DialogBackup
  </div>
  <div
    data-testid="dialog-confirm-backup"
  >
    DialogConfirmBackup
  </div>
</div>
`;

exports[`AlertBackup > matches snapshot for warning icon 1`] = `
<div
  class="px-6 py-3 bg-brand rounded-lg flex-row items-center gap-3"
  data-testid="container"
>
  <div
    class="flex-row flex-1 gap-3 items-center"
    data-testid="container"
  >
    <div
      class="h-4 w-4 font-bold text-primary-foreground"
      data-testid="triangle-alert"
    >
      TriangleAlert
    </div>
    <p
      class="font-bold text-primary-foreground whitespace-nowrap"
      data-size="sm"
      data-testid="typography"
    >
      Back up now
      <span
        class="hidden md:inline"
      >
        to avoid losing your account!
      </span>
    </p>
  </div>
  <div
    data-testid="dialog-backup"
  >
    DialogBackup
  </div>
  <div
    data-testid="dialog-confirm-backup"
  >
    DialogConfirmBackup
  </div>
</div>
`;

exports[`AlertBackup > matches snapshot for warning message 1`] = `
<div
  class="px-6 py-3 bg-brand rounded-lg flex-row items-center gap-3"
  data-testid="container"
>
  <div
    class="flex-row flex-1 gap-3 items-center"
    data-testid="container"
  >
    <div
      class="h-4 w-4 font-bold text-primary-foreground"
      data-testid="triangle-alert"
    >
      TriangleAlert
    </div>
    <p
      class="font-bold text-primary-foreground whitespace-nowrap"
      data-size="sm"
      data-testid="typography"
    >
      Back up now
      <span
        class="hidden md:inline"
      >
        to avoid losing your account!
      </span>
    </p>
  </div>
  <div
    data-testid="dialog-backup"
  >
    DialogBackup
  </div>
  <div
    data-testid="dialog-confirm-backup"
  >
    DialogConfirmBackup
  </div>
</div>
`;
